File: LCX/ValidEndPoints/dsl/examples/place_order.swift
import Foundation
import CommonCrypto


# POST method not supported for swift
func check_order_status(order_id, completion: @escaping (Result<[String: Any], Error>) -> Void) {
    // GET /api/trading/order/{order_id}
    let urlString = "https://exchange-api.lcx.com/api/trading/order/{order_id}"
    guard let url = URL(string: urlString) else {
        completion(.failure(NSError(domain: "Invalid URL", code: 0)))
        return
    }

    var request = URLRequest(url: url)
    request.addValue("application/json", forHTTPHeaderField: "Content-Type")
    let apiKey = ProcessInfo.processInfo.environment["LCX_API_KEY"] ?? ""
    let apiSecret = ProcessInfo.processInfo.environment["LCX_API_SECRET"] ?? ""
    let timestamp = String(Int(Date().timeIntervalSince1970 * 1000))
    let message = timestamp + "GET" + "/api/trading/order/{order_id}"

    var digest = [UInt8](repeating: 0, count: Int(CC_SHA256_DIGEST_LENGTH))
    message.withCString { messagePtr in
        _ = CC_HMAC(CCHmacAlgorithm(kCCHmacAlgSHA256), apiSecret, apiSecret.count, messagePtr, strlen(messagePtr), &digest, nil)
    }
    let signature = Data(digest).base64EncodedString()


    URLSession.shared.dataTask(with: request) { data, response, error in
        if let error = error {
            completion(.failure(error))
            return
        }

        guard let data = data else {
            completion(.failure(NSError(domain: "No data", code: 0)))
            return
        }
        if let json = try JSONSerialization.jsonObject(with: data) as? [String: Any] {{
            completion(.success(json))
        }}

    }.resume()
}

# POST method not supported for swift
func complete_order_flow(currency, completion: @escaping (Result<[String: Any], Error>) -> Void) {
    // GET /api/account/balance
    let urlString = "https://exchange-api.lcx.com/api/account/balance"
    guard let url = URL(string: urlString) else {
        completion(.failure(NSError(domain: "Invalid URL", code: 0)))
        return
    }

    var request = URLRequest(url: url)
    request.addValue("application/json", forHTTPHeaderField: "Content-Type")
    let apiKey = ProcessInfo.processInfo.environment["LCX_API_KEY"] ?? ""
    let apiSecret = ProcessInfo.processInfo.environment["LCX_API_SECRET"] ?? ""
    let timestamp = String(Int(Date().timeIntervalSince1970 * 1000))
    let message = timestamp + "GET" + "/api/account/balance"

    var digest = [UInt8](repeating: 0, count: Int(CC_SHA256_DIGEST_LENGTH))
    message.withCString { messagePtr in
        _ = CC_HMAC(CCHmacAlgorithm(kCCHmacAlgSHA256), apiSecret, apiSecret.count, messagePtr, strlen(messagePtr), &digest, nil)
    }
    let signature = Data(digest).base64EncodedString()


    URLSession.shared.dataTask(with: request) { data, response, error in
        if let error = error {
            completion(.failure(error))
            return
        }

        guard let data = data else {
            completion(.failure(NSError(domain: "No data", code: 0)))
            return
        }

    }.resume()
}



# POST method not supported for swift
# None method not supported for swift
# None method not supported for swift
# None method not supported for swift
# None method not supported for swift
# None method not supported for swift
# None method not supported for swift
# None method not supported for swift
# None method not supported for swift
# None method not supported for swift
# None method not supported for swift

File: LCX/ValidEndPoints/dsl/examples/place_order.php
<?php
error_reporting(E_ALL);
ini_set('display_errors', 1);


# POST method not supported for php
function check_order_status(order_id) {
    // GET /api/trading/order/{order_id}
    $url = "https://exchange-api.lcx.com/api/trading/order/{order_id}";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/trading/order/{order_id}";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    return json_decode($response, true);

}

# POST method not supported for php
function complete_order_flow(currency) {
    // GET /api/account/balance
    $url = "https://exchange-api.lcx.com/api/account/balance";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/account/balance";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }

}



# POST method not supported for php
# None method not supported for php
# None method not supported for php
# None method not supported for php
# None method not supported for php
# None method not supported for php
# None method not supported for php
# None method not supported for php
# None method not supported for php
# None method not supported for php
# None method not supported for php
